account_id.go

```diff
@@ -48,57 +48,62 @@ type _AccountIDs struct { //nolint
 // AccountIDFromString constructs an AccountID from a string formatted as
 // `Shard.Realm.Account` (for example "0.0.3")
 func AccountIDFromString(data string) (AccountID, error) {
-	shard, realm, num, checksum, alias, aliasEvmAddress, err := _AccountIDFromString(data)
+	shard, realm, num, checksum, alias, AliasEvmAddress, err := _AccountIDFromString(data)
 	if err != nil {
 		return AccountID{}, err
 	}
 
 	if num == -1 {
 		if alias != nil {
 			return AccountID{
-				Shard:           uint64(shard),
-				Realm:           uint64(realm),
-				Account:         0,
-				AliasKey:        alias,
+				Shard:      uint64(shard),
+				Realm:      uint64(realm),
+				Account:    0,
+				AliasKey:   alias,
 				AliasEvmAddress: nil,
-				checksum:        checksum,
+				checksum:   checksum,
 			}, nil
 		}
 
 		return AccountID{
-			Shard:           uint64(shard),
-			Realm:           uint64(realm),
-			Account:         0,
-			AliasKey:        nil,
-			AliasEvmAddress: aliasEvmAddress,
-			checksum:        checksum,
+			Shard:      uint64(shard),
+			Realm:      uint64(realm),
+			Account:    0,
+			AliasKey:   nil,
+			AliasEvmAddress: AliasEvmAddress,
+			checksum:   checksum,
 		}, nil
 	}
 
 	return AccountID{
-		Shard:           uint64(shard),
-		Realm:           uint64(realm),
-		Account:         uint64(num),
-		AliasKey:        nil,
+		Shard:      uint64(shard),
+		Realm:      uint64(realm),
+		Account:    uint64(num),
+		AliasKey:   nil,
 		AliasEvmAddress: nil,
-		checksum:        checksum,
+		checksum:   checksum,
 	}, nil
 }
 
-func AccountIDFromEvmAddress(shard uint64, realm uint64, evmAddress string) (AccountID, error) {
-	temp, err := hex.DecodeString(evmAddress)
+func AccountIDFromEvmAddress(shard uint64, realm uint64, aliasEvmAddress string) (AccountID, error) {
+	temp, err := hex.DecodeString(aliasEvmAddress)
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**suggestion:** ‏An additional check that the decoded value is 20 bytes can be added as well.